front/src/utils/debounce.ts

```diff
@@ -0,0 +1,10 @@
+import { MutableRefObject, useRef } from 'react';
+
+export function useDebouncedFunction<T extends Function>(func: T, delay = 300) {
+  const ref: MutableRefObject<number | undefined> = useRef(undefined);
```

Comment:
@tadite This looks perfect. One question - why is the timeout stored in a ref?

Reply:
Because in `SearchPage.tsx` `useDebouncedFunction` called from react functional component `const SearchPage`, so we should store ref in component scope. Without useRef it will create new ref every time React calls component and debounce wont work.

Reply:
Thank you - this looks great.